python/tank/util/shotgun.py

```diff
@@ -144,6 +136,21 @@ def __get_sg_config_data(shotgun_cfg_path, user="default"):
     except Exception, error:
         raise TankError("Cannot load config file '%s'. Error: %s" % (shotgun_cfg_path, error))
 
+    return _parse_config_data(file_data, user, shotgun_cfg_path)
+
+
+def _parse_config_data(file_data, user, shotgun_cfg_path):
+    """
+    Parses configuration data and overrides it with the studio level hook's result if available.
```

Comment:
I would love to see an example of what one of these files are expected to look like (one example for each supported format). I personally find that looking at a clear example of a config file is very clear and help me understand what the code is expecting.